src/main/java/io/fabric8/maven/docker/util/AnsiLogger.java

```diff
@@ -59,44 +62,78 @@ public AnsiLogger(Log log, boolean useColor, String verbose) {
     }
 
     public AnsiLogger(Log log, boolean useColor, String verbose, boolean batchMode, String prefix) {
+        this(log, useColor, verbose, batchMode, prefix, null);
+    }
+
+    public AnsiLogger(Log log, boolean useColor, String verbose, boolean batchMode, String prefix, File outpufFile) {
         this.log = log;
         this.prefix = prefix;
-        this.batchMode = batchMode;
+        this.outputFile = outpufFile;
+        if (this.outputFile == null) {
+            this.batchMode = batchMode;
+        } else {
+            this.batchMode = true;
+        }
         checkVerboseLoggingEnabled(verbose);
         initializeColor(useColor);
+        try {
+            initializePrintWriter();
+        } catch (FileNotFoundException e) {
+            e.printStackTrace();
+        }
     }
 
     /** {@inheritDoc} */
     public void debug(String message, Object ... params) {
         if (isDebugEnabled()) {
-            log.debug(prefix + format(message, params));
+            if (outputFile != null) {
+                pw.println(format(message, params));
+            } else {
+                log.debug(prefix + format(message, params));
+            }
         }
     }
 
     /** {@inheritDoc} */
     public void info(String message, Object ... params) {
-        log.info(colored(message, COLOR_INFO, true, params));
+        if (outputFile != null && log.isInfoEnabled()) {
+            pw.println(format(message, params));
+        } else {
+            log.info(colored(message, COLOR_INFO, true, params));
+        }
     }
 
     /** {@inheritDoc} */
     public void verbose(LogVerboseCategory logVerboseCategory, String message, Object ... params) {
         if (isVerbose && verboseModes != null && verboseModes.contains(logVerboseCategory)) {
-            log.info(ansi().fgBright(BLACK).a(prefix).a(format(message, params)).reset().toString());
+            if (outputFile != null) {
+                pw.println(format(message, params));
+            } else {
+                log.info(ansi().fgBright(BLACK).a(prefix).a(format(message, params)).reset().toString());
+            }
         }
     }
 
     /** {@inheritDoc} */
-    public void warn(String format, Object ... params) {
-        log.warn(colored(format, COLOR_WARNING, true, params));
+    public void warn(String message, Object ... params) {
+        if (outputFile != null && log.isWarnEnabled()) {
+            pw.println(format(message, params));
+        } else {
+            log.warn(colored(message, COLOR_WARNING, true, params));
+        }
     }
 
     /** {@inheritDoc} */
     public void error(String message, Object ... params) {
-        log.error(colored(message, COLOR_ERROR, true, params));
+        if (outputFile != null && log.isErrorEnabled()) {
+            pw.println(format(message, params));
+        } else {
+            log.error(colored(message, COLOR_ERROR, true, params));
+        }
```

Comment:
Do you think maybe we should move this if-else block to a method? Maybe something like this:
```java
private void logOrPrintToFile(Predicate<Log> logPredicate, Consumer<Log> logConsumer, String message, Object ... params) {
    if (outputFile != null && logPredicate.test(log)) {
        pw.println(format(message, params));
    } else {
        logConsumer.accept(log);
    }
}
```

Reply:
I have still to practice some more functional Java 👍 🙂 
I like it! thanks for the suggestion!